server/index.js

```diff
@@ -33,7 +33,12 @@ export default class Server {
     this.http = null
     this.config = getConfig(this.dir, conf)
     this.dist = this.config.distDir
-    this.buildStats = !dev ? require(join(this.dir, this.dist, 'build-stats.json')) : null
+    try {
+      this.buildStats = !dev ? require(join(this.dir, this.dist, 'build-stats.json')) : null
+    } catch (e) {
+      console.error('No build-stats.json found. Did you forget to run `next build`?')
```

Comment:
Looks like `next start` is already checking if a build exists, but the code does the checking is after trying to create a server. I put them in the correct order in my last commit.